butlarr/services/bazarr.py

```diff
@@ -0,0 +1,313 @@
+        
+        items = self.search(arr_variant=arr_variant, id=media_id)  
+
+        state = State(
+            items=items,
+            index=0,
+            arr_variant=arr_variant,
+            media_id=media_id,
+            menu="list",
+            parent=kwargs.get('parent')
+        )
+
+        self.session_db.add_session_entry(
+            default_session_state_key_fn(self, update), state
+        )
+
+        auth_level = get_auth_level_from_message(self.db, update)
+        allow_edit = auth_level >= AuthLevels.USER.value
+        return self.create_message(state, full_redraw=False, allow_edit=allow_edit)
+
+    @repaint
+    @callback(cmds=["download"])
+    @sessionState(clear=True)
+    @authorized(min_auth_level=AuthLevels.USER)
+    async def clbk_add(self, update, context, args, state):
+
+        logger.debug(f"Return to menu {state.parent.state.menu}")
+
+        state = replace(
+                    state,
+                    index=0,
+                    menu="success",
+                )
+
+        result = self.download(
+            id = state.media_id,
+            service=state.arr_variant,
+            item=state.items[state.index]
+        )
+
+        if result.status_code < 200 \
+        or result.status_code > 299:
+            return Response(caption=f"Something went wrong... {result.content}")
+        
+        return self.create_message(state, full_redraw=False)
+    
+    @Addon.config
+    def addon_buttons(self, **kwargs):
+        parent_state = self.state
+        item = parent_state.items[parent_state.index]
+        in_library = "id" in item and item["id"]
+
+        buttons = []
+        if ArrVariant(self.service.arr_variant) == ArrVariant.RADARR:
+            downloaded = True if "movieFile" in item else False
+
+            if parent_state.menu == "add" and downloaded:
+                movieId = item.get("id")
+                
+                buttons.append(
+                    Button(
+                        f"Search for subtitles",
+                        self.get_clbk("list", movieId)
+                    ),
+                )
+
+        elif ArrVariant(self.service.arr_variant) == ArrVariant.SONARR:
+
+            if parent_state.menu == "add" and in_library:
+                buttons.append(
+                    Button(
+                        f"Get Subtitles",
+                        self.service.get_clbk("seasons")
+                    ),
+                )
+
+            elif parent_state.menu == "episode" and in_library:
+
+                episodeId = item['selectedEpisodeId']
+                episode = self.service.get_episode(episodeId)
+                downloaded = True if episode['hasFile'] else False
+
+                if downloaded:
+                    buttons.append(
+                        Button(
+                            f"Search for subtitles",
+                            self.get_clbk("list", episodeId)
+                        ),
+                    )
```

Comment:
Same here, use the Addon interface to force an implementation for this :)

Reply:
I moved it tho service especific functions, so it won't be as ugly but it's still inside bazarr.py for now